### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by smacker**
Fixes https://github.com/src-d/code-annotation/issues/97

* Set log level depends on env and pass logger around correctly
* New vendor dependency

### CODE REVIEW

**Review by bzz — APPROVED** [review]
LGTM.

Rebase seems to be in order though.

File: server/service/logger.go [hunk 1]
```diff
@@ -5,12 +5,16 @@ import (
...
 // NewLogger returns a logrus Logger
-func NewLogger() logrus.FieldLogger {
+func NewLogger(env string) *logrus.Logger {
 	logger := logrus.New()
 	logger.Formatter = &logrus.TextFormatter{
 		FullTimestamp:   true,
 		TimestampFormat: "2006-01-02 15:04:05",
 	}
 
+	if env == "production" || env == "staging" {
```
Comment (dpordomingo) [inline server/service/logger.go]: `production` behavior should be the default one.
It's better to override things when we're in development.

**Review by dpordomingo — CHANGES_REQUESTED** [review]
mostly lgtm but one thing about the defaults...

File: server/service/logger.go (hunk 1 above)
Comment (smacker) [reply server/service/logger.go]: I mimic `yarn/npm` behavior. But production by default is also ok for me. @bzz what do you think?

Comment (dpordomingo) [reply server/service/logger.go]: We're changing things to make the app "prod first"... so I'm not sure if it's a good idea to develop new things in the other way...

Comment (bzz) [reply server/service/logger.go]: 👍  for production-first approach

**Comment by smacker** [comment]
changed to prod first approach

**Review by dpordomingo — APPROVED** [review]
LGTM....

I hope you only changed this https://github.com/src-d/code-annotation/pull/102/files#diff-3e5785ed7dfffb946332460b8a47b99dR15
Otherwise... idk

(commits in PRs being reviewed should not be amended; otherwise, the reviewer can not review (only) the changes)

**Comment by smacker** [comment]
Default has been changed from `dev` to `production` and env variable is removed from helm chart.
If you skip vendoring -  there are just 16 lines of code have been changed in total (yes, I counted).

**Comment by bzz** [comment]
👍  Shall we merge this guy then?

**Pull request merged**